src/ng/filter/orderBy.js

```diff
@@ -177,7 +177,7 @@ orderByFilter.$inject = ['$parse'];
 function orderByFilter($parse) {
   return function(array, sortPredicate, reverseOrder) {
 
-    if (!(isArrayLike(array))) return array;
+    if (!(isArrayLike(array))) throw minErr('orderBy')('notarray', 'Expected array but received: {0}', array);
```

Comment:
I was in the middle of merging when I realized there is an issue here: We need to allow `null`/`undefined` (in order to support asnc loading). 
This block needs to be modified (see [filter#L132-L138][1] for an example).

[1]: https://github.com/angular/angular.js/blob/3ca4ca463c97a2bbca09f0366d2a753a13751143/src/ng/filter/filter.js#L132-L138